src/libs/commonlib.cpp

```diff
@@ -802,33 +883,96 @@ Namespace* CoreIRLoadLibrary_commonlib(Context* c) {
       Type* out_type  = genargs.at("output_type")->get<Type*>();
+      Type* img_type = genargs.at("image_type")->get<Type*>();
+
+      // process and check the input arguments
+      vector<uint> in_dims = get_dims(in_type);
+      vector<uint> out_dims = get_dims(out_type);
+      vector<uint> img_dims = get_dims(img_type);
+
+      uint bitwidth = in_dims[0]; // first array is bitwidth
+      ASSERT(bitwidth > 0, "The first dimension for the input is interpretted "
+             "as the bitwidth which was set to " + to_string(bitwidth));
+
+      ASSERT(bitwidth == out_dims[0], 
+             to_string(bitwidth) + " != " + to_string(out_dims[0]) + \
+             "all bitwidths must match (input doesn't match output)");
+      ASSERT(bitwidth == img_dims[0], 
+             to_string(bitwidth) + " != " + to_string(img_dims[0]) + \
+             "all bitwidths must match (input doesn't match image)");
+
+      // erase the bitwidth size from vectors
+      in_dims.erase(in_dims.begin()); 
+      out_dims.erase(out_dims.begin());
+      img_dims.erase(img_dims.begin());
+
+      uint num_dims = in_dims.size();
+      ASSERT(num_dims == out_dims.size(),
+             "all must have same number of dimensions (input and output mismatch)");
+      ASSERT(num_dims == img_dims.size(),
+             "all must have same number of dimensions (input and image mismatch)");
+
+      // we will check all dimensions for correct construction
+      for (uint dim=0; dim<num_dims; ++dim) {
+        uint out_dimx = out_dims[dim];
+        uint img_dimx = img_dims[dim];
+        uint in_dimx = in_dims[dim];
+
+        ASSERT(img_dimx >= out_dimx,
+               "image dimension length (" + to_string(img_dimx) + \
+               ") must be larger than output (" + to_string(out_dimx) + \
+               ") in dim " + to_string(dim));
+        ASSERT(out_dimx >= in_dimx,
+               "output stencil size (" + to_string(out_dimx) + \
+               ") must be larger than input (" + to_string(in_dimx) + \
+               ") in dim " + to_string(dim));
+        ASSERT(img_dimx % in_dimx == 0,
+               "img_dim=" + to_string(img_dimx) + " % in_dim=" + to_string(in_dimx) + \
+               " != 0 in dim=" + to_string(dim) + \
+               ", dimension length must be divisible, because we can't swizzle data");
+        ASSERT(out_dimx % in_dimx == 0,
+               "out_dim=" + to_string(out_dimx) + " % in_dim=" + to_string(in_dimx) + \
+               " != 0 in dim=" + to_string(dim) + \
+               ", dimension length must be divisible, because we can't swizzle data");
+
+        if (img_dimx - out_dimx < 3 && (img_dimx != out_dimx)) {
+          std::cout << "Image dimension " << dim << "  is " << img_dimx
+                    << " and output stencil size is " << out_dimx
+                    << ", which means the linebuffer mem is going to be very small"
+                    << std::endl;
+      
+        }
+      }
```

Comment:
Can you separate this for loop into a helper function? (such as `check_dimensions_are_correct`), that would help make this code more organized.